.github/workflows/e2e-with-binary.yml

```diff
@@ -64,3 +64,35 @@ jobs:
             echo "file does not exist, or is empty"
             exit 1
           fi
+  e2e-tests-with-sget:
+    # Skip if running in a fork that might not have secrets configured.
+    if: ${{ github.repository == 'sigstore/cosign' }}
+    name: Run tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ macos-latest, ubuntu-latest, windows-latest ]
+
+    permissions:
+      id-token: write
+      contents: read
+    env:
+      COSIGN_EXPERIMENTAL: "true"
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.17.x'
+      - name: build cosign and check
```

Comment:
It might make sense to expand this test with more flag combinations, but I am not sure if this is really necessary.

We should also test the image subcommand.